content/learn/book/migration-guides/0.9-0.10/_index.md

```diff
@@ -0,0 +1,733 @@
+</div>
+
+- Add a `mut` for `removed: RemovedComponents<T>` since we are now modifying an event reader internally.
+- Iterating over removed components now requires `&mut removed_components` or `removed_components.iter()` instead of `&removed_components`.
+
+### [Remove `ExclusiveSystemParam::apply`](https://github.com/bevyengine/bevy/pull/7489)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+_Note for maintainers: this migration guide makes more sense if it’s placed above the one for #6919._
+
+The trait method `ExclusiveSystemParamState::apply` has been removed. If you have an exclusive system with buffers that must be applied, you should apply them within the body of the exclusive system.
+
+### [add `UnsafeWorldCell` abstraction](https://github.com/bevyengine/bevy/pull/6404)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+<!-- TODO -->
+
+### [Added `resource_id` and changed `init_resource` and `init_non_send_resource` to return `ComponentId`](https://github.com/bevyengine/bevy/pull/7284)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+<!-- TODO -->
+
+### [Basic adaptive batching for parallel query iteration](https://github.com/bevyengine/bevy/pull/4777)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+The `batch_size` parameter for `Query(State)::par_for_each(_mut)` has been removed. These calls will automatically compute a batch size for you. Remove these parameters from all calls to these functions.
+
+Before:
+
+```rust
+fn parallel_system(query: Query<&MyComponent>) {
+   query.par_for_each(32, |comp| {
+        ...
+   });
+}
+```
+
+After:
+
+```rust
+fn parallel_system(query: Query<&MyComponent>) {
+   query.par_iter().for_each(|comp| {
+        ...
+   });
+}
+```
+
+### [Support piping exclusive systems](https://github.com/bevyengine/bevy/pull/7023)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+Exclusive systems (systems that access `&mut World`) now support system piping, so the `ExclusiveSystemParamFunction` trait now has generics for the `In`put and `Out`put types.
+
+```rust
+// Before
+fn my_generic_system<T, Param>(system_function: T)
+where T: ExclusiveSystemParamFunction<Param>
+{ ... }
+
+// After
+fn my_generic_system<T, In, Out, Param>(system_function: T)
+where T: ExclusiveSystemParamFunction<In, Out, Param>
+{ ... }
+```
+
+### [Document alignment requirements of `Ptr`, `PtrMut` and `OwningPtr`](https://github.com/bevyengine/bevy/pull/7151)
+
+<div class="migration-guide-area-tags">
+    <div class="migration-guide-area-tag">ECS</div>
+</div>
+
+- Safety invariants on `bevy_ptr` types’ `new` `byte_add` and `byte_offset` methods have been changed. All callers should re-audit for soundness.

```

Comment:
Breaking changes to safety invariants seem like they could insidiously hard to find, and the consequences result in UB. Should we show some higher priority to these?

Reply:
Right now, the only sort is by area and then by close date. I have no issue with changing ordering, although, I think this should keep the area ordering. So in this case, keep it close to other ECS related guides. As mentioned in another comment, I'd prefer waiting until closer to release before actually moving it so I'll leave the conversation opened.